src/Microsoft.AspNetCore.Mvc.Core/Internal/FilterActionInvoker.cs

```diff
@@ -336,18 +337,6 @@ private Task InvokeAllResourceFiltersAsync()
                 }
                 else
                 {
-                    // We've reached the end of resource filters, so move to setting up state to invoke model
-                    // binding.
-                    var valueProviders = new List<IValueProvider>();
-                    var factoryContext = new ValueProviderFactoryContext(Context);
-
-                    for (var i = 0; i < _valueProviderFactories.Count; i++)
-                    {
-                        var factory = _valueProviderFactories[i];
-                        await factory.CreateValueProviderAsync(factoryContext);
-                    }
-                    Context.ValueProviders = factoryContext.ValueProviders;
```

Comment:
Answering my own question: This makes a difference when no controller properties or action parameters need to be bound. Also opens the door on avoiding the factories in additional cases e.g. the all-singing all-dancing all-`[FromBody]` case.